core/trino-main/src/main/java/io/trino/operator/AggregationOperator.java

```diff
@@ -176,4 +169,21 @@ public Page getOutput()
         state = State.FINISHED;
         return pageBuilder.build();
     }
+
+    private boolean updateMemory()
+    {
+        long memorySize = 0;
+        for (Aggregator aggregate : aggregates) {
+            memorySize += aggregate.getEstimatedSize();
+        }
+        if (useSystemMemory) {
+            systemMemoryContext.setBytes(memorySize);
+        }
+        else {
+            userMemoryContext.setBytes(memorySize);
+        }
+
+        // This is needed to be fit into UpdateMemory.
```

Comment:
Actually, we can't we return future from `setBytes` here?

Reply:
Discussion on-going in Slack